lib/Test/Pod/CoverageChange.pm

```diff
@@ -0,0 +1,246 @@
+package Test::Pod::CoverageChange;
+# ABSTRACT: check perl modules against their pod coverage
+
+=pod
+
+=head1 NAME
+
+Pod coverage calculator test file.
+
+=head1 SYNOPSIS
+
+    use Test::Pod::CoverageChange qw(check);
+
+    Test::Pod::CoverageChange::check('lib', {
+        Module::With::3::expected::naked::subs              => 3,
+        AnotherModule::With::10::expected::naked::subs      => 10,
+        YetAnotherModule::With::1::expected::naked::subs    => 1,
+        YetAnotherModule::With::5::expected::naked::subs    => 5,
+    }, [
+        We::Ignore::ThisModule,
+        We::Also::Ignore::This::Module
+    ]);
+
+=head1 DESCRIPTION
+
+It checks all files that placed under a given directory against their
+POD coverage to see if all existing subs have POD and also
+POD syntax to see if is there an POD syntax error or not?
+
+Prints the percentage of POD coverage in B<TODO> test format for the packages that we allowed to have naked subs.
+Prints an error message if our latest changes increased/decreased numbers of naked sub for the packages that we allowed to have naked sub.
+Prints an error message if a naked allowed package has 100% POD coverage. (We should remove it from the C<%naked_packages> variable list.)
+Ignores to check every package that we pass as C<@ignored_package>
+Prints a proper message for the newly added packages.
+
+Prints C<ok> for the files that have no POD syntax error.
+Prints C<not ok- There is no POD in the file> if the file has no POD at all. I will consider as a pass test since it is in a todo section.
+Prints C<not ok- The number of errors in the POD structure> if the file has any error. It causes tests to be fail.
+
+=cut
+
+use strict;
+use warnings;
+
+use Test::More;
+use Pod::Checker;
+use Pod::Coverage;
+use File::Find::Rule;
+use Test::Pod::Coverage;
+use Module::Path 'module_path';
+use List::Util qw(any);
+
+our $VERSION = '0.01';
+
+use constant {
+    POD_SYNTAX_IS_OK => 0,
+    FILE_HAS_NO_POD  => -1,
+};
+
+use Exporter qw(import export_to_level);
+our @EXPORT_OK = qw(check);
+
+# Set caller test file name
+my $caller_test_file = (caller())[1];
+
+# Create a test builder object
+my $Test_Builder = Test::More->builder;
+
+=pod
+
+=head2 check
+
+Check all modules under a given directory against POD coverage and POD syntax
+
+=over 4
+
+=item * C<directories> - directories to check recursively, supports string or arrayref
```

Comment:
inconsistent, doesn't match the naming in the code and misleading because it looks like these are named parameters...

```suggestion
=item * C<$path> - path or arrayref of directories to check (recursively)
```